generators/upgrade/index.js

```diff
@@ -0,0 +1,83 @@
+const Base = require('yeoman-generator').Base;
+const downloadFileSync = require('download-file-sync');
+require('colors');
+const versions = require('./versions.js');
+
+const baseAppName = 'RnDiffApp';
+const baseLowercaseAppName = 'rndiffapp';
+const patchFileName = 'upgrade-rn.patch';
+
+class UpgradeGenerator extends Base {
+
+  initializing() {
+    this.conflicter.force = true;
+    const config = this.fs.readJSON(this.destinationPath('package.json'));
+
+    if (!config.dependencies || !config.dependencies['react-native']) {
+      throw new Error('React Native dependency not detected');
+    }
+
+    const rnVersionId = config.dependencies['react-native'].replace(/[\^~]/g, '');
+    const rnVersionIndex = versions.findIndex(version => (version.id === rnVersionId));
+
+    if (rnVersionIndex === -1) {
+      console.log(`Your React Native version (${rnVersionId}) is not compatible with this tool`.red);
```

Comment:
Could use `this.log.error` instead of `color` but that does not bother me. Up to you to change it if you want